src/smc-webapp/jupyter/parsing.ts

```diff
@@ -35,3 +36,25 @@ function last_style(code: string, mode = "python") {
   CodeMirror.runMode(code, mode, (_, s) => (style = s));
   return style;
 }
+
+// calling escape and then unescape should be idempotent
```

Comment:
In this comment, you're not using the mathematical/English word "idempotent" correctly.  It means an element of a group that is equal to its square, e.g., x*x = x.   What you mean is that the composition of escape followed by unescape is the identity map, which is just a completely different thing.